pyroute2/fixtures/iproute.py

```diff
@@ -0,0 +1,523 @@
+        address=test_link_ifinfmsg.get('address'),
+        nsname=nsname,
+    )
+
+
+@pytest.fixture(name='test_link_address')
+def _test_link_address(test_link: TestInterface) -> Generator[str]:
+    '''Test interface MAC address.
+
+    * **Name**: test_link_address
+    * **Scope**: function
+    * **Depends**: test_link
+
+    Yield test interface MAC address. The network namespace and
+    the test interface exist at this point.
+    '''
+    yield test_link.address
+
+
+@pytest.fixture(name='test_link_index')
+def _test_link_index(test_link: TestInterface) -> Generator[int]:
+    '''Test interface MAC index.
+
+    * **Name**: test_link_index
+    * **Scope**: function
+    * **Depends**: test_link
+
+    Yield test interface index. The network namespace and
+    the test interface exist at this point.
+    '''
+    yield test_link.index
+
+
+@pytest.fixture(name='test_link_ifname')
+def _test_link_ifname(test_link: TestInterface) -> Generator[str]:
+    '''Test interface MAC ifname.
+
+    * **Name**: test_link_ifname
+    * **Scope**: function
+    * **Depends**: test_link
+
+    Yield test interface ifname. The network namespace and
+    the test interface exist at this point.
+    '''
+    yield test_link.ifname
+
+
+@pytest.fixture(name='tmp_link_ifname')
+def _tmp_link_ifname(nsname: str) -> Generator[str]:
+    '''Temporary link name.
+
+    * **Name**: tmp_link_ifname
+    * **Scope**: function
+    * **Depends**: nsname
+
+    Yield tmp link ifname, but don't create it. Try to remove
+    the link on cleanup.
+    '''
+    ifname = uifname()
+    with IPRoute(netns=nsname) as ipr:
+        yield ifname
+        try:
+            (link,) = ipr.link('get', ifname=ifname)
+            ipr.link('del', index=link.get('index'))
+        except NetlinkError as e:
+            if e.code != errno.ENODEV:
+                raise
+
+
+@pytest_asyncio.fixture(name='async_ipr')
+async def _async_ipr(request, nsname: str) -> AsyncGenerator[AsyncIPRoute]:
+    '''`AsyncIPRoute` instance.
+
+    * **Name**: async_ipr
+    * **Scope**: function
+    * **Depends**: nsname
+
+    Yield `AsyncIPRoute` instance, running within the test network namespace.
+    You can provide additional keyword arguments to `AsyncIPRoute`:
+
+    .. testcode::
+        :hide:
+
+        import pytest
+        from unittest.mock import patch
+
+        patcher = patch("pytest.mark")
+        patcher.start()
+
+    .. testcode::
+
+        @pytest.mark.parametrize(
+            'async_ipr',
+            [
+                {
+                    'ext_ack': True,
+                    'strict_check': True,
+                },
+            ],
+            indirect=True
+        )
+        @pytest.mark.asyncio
+        async def test_my_case(async_ipr):
+            await async_ipr.link('set', index=1, state='up')
+
+    .. testcode::
+        :hide:
+
+        patcher.stop()
+
+        import asyncio
+
+        async def test_wrapper(calls):
```

Comment:
this function clearly needs a bit of explaining imho :)

Reply:
That's also an assertion code, not an example.

Yep, you have convinced me, I'll rearrange the assertion code so it will not be visible in the docstrings.